### PR TITLE
Some cleanups for the codebase

### PR DESCRIPTION
**Pull request by ikesyo**
See each commit for the details.

### CODE REVIEW

File: Commandant/ArgumentParser.swift [hunk 1]
```diff
@@ -125,7 +125,11 @@ public final class ArgumentParser {
 		rawArguments.removeAll()
 
 		var foundValue: String?
-		argumentLoop: for var index = 0; index < oldArguments.count; index++ {
+		var index = 0
+
+		while index < oldArguments.count {
+			defer { index++ }
```
Comment (ikesyo) [inline Commandant/ArgumentParser.swift]: We should avoid `++` too: https://github.com/apple/swift-evolution/blob/master/proposals/0004-remove-pre-post-inc-decrement.md

File: Commandant/Command.swift [hunk 2]
```diff
@@ -151,10 +152,10 @@ extension CommandRegistry {
 		}
 
 		switch runCommand(verb, arguments: arguments) {
-		case .Some(.Success):
+		case (.Success?):
 			exit(EXIT_SUCCESS)
 
-		case let .Some(.Failure(error)):
+		case let (.Failure(error)?):
```
Comment (mdiep) [inline Commandant/Command.swift]: Do we need the outer parens here? `case let .Failure(error)?:`?

**Comment by mdiep** [comment]
Just the one question. :+1:

File: Commandant/Command.swift (hunk 2 above)
Comment (ikesyo) [inline Commandant/Command.swift]: You are right! :white_check_mark:

**Comment by ikesyo** [comment]
Just fixed. :zap:

File: Commandant/Command.swift [hunk 3]
```diff
@@ -151,10 +152,10 @@ extension CommandRegistry {
 		}
 
 		switch runCommand(verb, arguments: arguments) {
-		case .Some(.Success):
+		case .Success?:
```
Comment (NachoSoto) [inline Commandant/Command.swift]: Whoa TIL

**Comment by mdiep** [comment]
:zap:

**Pull request merged**